### ISSUE TITLE
Automation account datasource is inverting resource group name and account name when searching the resource in Azure

### ISSUE THREAD

**Issue by xadozuk** [hashicorp/terraform-provider-azurerm#14463]
## Terraform (and AzureRM Provider) Version
Terraform version 0.14.11, AzureRM version 2.88.0

## Affected Resource(s)
azurerm_automation_account (datasource)

## Expected Behaviour
Using a datasource for an already created Automation account should find it in Azure.

## Steps to Reproduce
Using a datasource for an already existing automation account will throw a not found error:
```
Automation Account: (Name "<resource group name>" / Resource Group "<automation account name>") was not found
```

## Possible issue found in code

Resource group and name seems to be inverted

https://github.com/hashicorp/terraform-provider-azurerm/blob/d65781db71babf989c7e0ae17964c839af6acc90/internal/services/automation/automation_account_data_source.go#L54

Function definition:
https://github.com/hashicorp/terraform-provider-azurerm/blob/d65781db71babf989c7e0ae17964c839af6acc90/internal/services/automation/parse/automation_account.go#L18

**Comment by aidanfinn** [comment]
I'm seeing issues with Automation Account too. Using the docs example I get:

Error: creating/updating Automation Account: (Name "example-resources" / Resource Group "account1"): automation.AccountClient#CreateOrUpdate: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceGroupNotFound" Message=**"Resource group 'account1'** could not be found."
│
│   with azurerm_automation_account.example,
│   on test.tf line 10, in resource "azurerm_automation_account" "example":
│   10: resource "azurerm_automation_account" "example" {

It appears that the name property is incorrectly used as the resource group name:

resource "azurerm_resource_group" "example" {
  **name     = "example-resources"**
  location = "West Europe"
}

resource "azurerm_automation_account" "example" {
  **name                = "account1"**
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  sku_name = "Basic"
}

**Comment by xadozuk** [comment]
I created a pull request with the supposed change (if it is confirmed).
First time contributor, let me know if I need to provide more information.

**Comment by stephybun** [comment]
Closed by #14464
